chall01/hopham.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/python
+
+import sys
+import string
+
+morse = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", 
+        ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", 
+        "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
+
+if (len(sys.argv) != 2 or len(sys.argv[1]) == 0):
+    print("usage: ./hopham.py <a-zA-Z string>")
+    exit()
+argument = sys.argv[1]
+argument_lowcase = argument.lower()
+for character in argument_lowcase:
+    if (character not in string.ascii_lowercase and character != " "):
+        print("usage: ./hopham.py <a-zA-Z string>")
+        exit()
+
+morse_string = dict()
+i = 0
+for key in string.ascii_lowercase:
+    morse_string[key] = morse[i]
+    i += 1
```

Comment:
the pythonic way to do the same thing is:

```python
morse_string = {k: v for k, v in zip(string.ascii_lowercase, morse)}
```

Using a dict was a really good idea!